### PR TITLE
feat(avatar): Port avatar name-color logic from lifeguide (DS-1623)

### PR DESCRIPTION
**Pull request by toby-daigle**
Screenshot 2026-07-31 at 11 17 59

### CODE REVIEW

File: packages/react/src/components/avatar/avatar.test.tsx [hunk 1]
```diff
@@ -122,4 +123,70 @@ describe('Avatar', () => {
...
+            const containerAccented = renderWithProviders(<Avatar username="Éric" />).container;
+            const containerUnaccented = renderWithProviders(<Avatar username="Eric" />).container;
+
+            const elementAccented = containerAccented.firstChild as HTMLElement;
+            const elementUnaccented = containerUnaccented.firstChild as HTMLElement;
+
+            expect(elementAccented).toHaveStyle({ backgroundColor: AZ_BG_COLORS[4] });
+            expect(elementUnaccented).toHaveStyle({ backgroundColor: AZ_BG_COLORS[4] });
+        });
+    });
```
Comment (toby-daigle) [inline packages/react/src/components/avatar/avatar.test.tsx:191]: There are missing test cases for when the username is empty or undefined. It should use the theme fallback.

**Review by toby-daigle — COMMENTED** [review]
RESOLVED

1. Test snap
This PR did not update the snapshot ( file avatar.test.tsx.snap ) so i expect some tests to fail.

File: packages/react/src/components/avatar/avatar.tsx [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import { type FC, useMemo } from 'react';
-import styled, { css, FlattenInterpolation, ThemeProps } from 'styled-components';
+import styled, { css, FlattenInterpolation, ThemeProps, useTheme } from 'styled-components';
```
Comment (pylafleur) [inline packages/react/src/components/avatar/avatar.tsx]: C'est pas le bon import, utilise `import { useTheme } from '../../hooks/use-theme';` et t'auras pas à utiliser le type hint de `ResolvedTheme` plus bas.

**Review by pylafleur — COMMENTED** [review]
Juste l'import à corriger, le reste est ben beau!

C'est nice, on avait copié-collé ce code-là dans /connect, on va pouvoir l'utiliser aussi!

**Review by pylafleur — APPROVED** [review]

**Comment by toby-daigle** [comment]
@pylafleur  j'ai fix le commit message, one more review please 👍

**Review by pylafleur — APPROVED** [review]

File: packages/react/src/components/user-profile/user-profile.test.tsx.snap
Comment (meriouma) [inline packages/react/src/components/user-profile/user-profile.test.tsx.snap:1]: @toby-daigle Je pense pas que la couleur dans le UserProfile devait changer. Celle-là doit être fixe il me semble.

Comment (pylafleur) [reply packages/react/src/components/user-profile/user-profile.test.tsx.snap:1]: Je me disais que ça serait plus cohérent s'il était de la même couleur dans le header et dans le contenu. C'est la même information en réalité. Et avec le header pâle qui s'en vient, ça va ajouter une touche de couleur! 😄 

Mais si oui, faudrait mettre à jour le figma.

Comment (toby-daigle) [reply packages/react/src/components/user-profile/user-profile.test.tsx.snap:1]: @meriouma  // @pylafleur 

Pour addresser ce point, j'ai parlé à Laury et créé un carte Jira et une PR.

https://equisoft.atlassian.net/browse/DS-1639
https://github.com/kronostechnologies/design-elements/pull/1514
